File: uservice-reports/main/src/test/scala/sinnet/ReportService2Test.scala
package sinnet

import io.quarkus.test.junit.QuarkusTest
import io.restassured.RestAssured.given
import org.hamcrest.CoreMatchers.`is`
import org.junit.jupiter.api.Test
import io.quarkus.example.GreeterGrpc;
import io.quarkus.example.HelloRequest;
import javax.inject.Inject
import scala.annotation.meta.field
import sinnet.reports.{YearMonth => YearMonthDTO}
import sinnet.reports.report2.{
        ReportRequest => ReportRequestDTO,
        ActivityDetails => ActivityDetailsDTO}
import sinnet.reports.report2.ReportsGrpc
import org.assertj.core.api.Assertions
import java.util.zip.ZipInputStream
import java.io.ByteArrayInputStream
import io.quarkus.grpc.GrpcClient
import org.junit.jupiter.api.Timeout

@QuarkusTest
@Timeout(3)
class ReportService2Test {
    
    @Inject
    @GrpcClient
    var self: ReportsGrpc.ReportsBlockingStub = _

    @Test
    def produceReportWithMinDataCase1(): Unit = {
        val activity = ActivityDetailsDTO.newBuilder().build()
        val request = ReportRequestDTO.newBuilder().build()
        val res = self.produce(request)
        var data = res.getData().toByteArray()
        Assertions
            .assertThat(data)
            .isNotEmpty()
    }

    @Test
    def produceReportWithMinDataCase2(): Unit = {
        var period = YearMonthDTO.newBuilder().setYear(2001).setMonth(1);
        val request = ReportRequestDTO.newBuilder()
            .addDetails(ActivityDetailsDTO
                .newBuilder()
                .setPersonName("Ala")
                .setYearMonth(period)
                .build())
            .addDetails(ActivityDetailsDTO
                .newBuilder()
                .setPersonName("Ola")
                .setYearMonth(period)
                .build())
            .build()
        val res = self.produce(request)
        var data = res.getData().toByteArray()
        
        // uncomment block of lines below to produce a local example raport file
        import java.io.File
        import java.nio.file.Files
        import java.nio.file.Paths
        Files.write(Paths.get("temp_raport2_from_test.pdf"), data)

        Assertions.assertThat(data).isNotEmpty()
    }

    // @Test
    // def producePackEndpoint(): Unit = {
    //     val request = ReportRequestDTO.newBuilder().build
    //     val pack = ReportRequestsDTO.newBuilder()
    //         .addItems(request)
    //         .addItems(request)
    //         .build()
    //     val res = self.producePack(pack)

    //     var data = res.getData().toByteArray()
    //     val byteStream = new ByteArrayInputStream(data)
    //     val zis = new ZipInputStream(byteStream)

    //     var entry1 = zis.getNextEntry()
    //     var entry2 = zis.getNextEntry()
    //     var entry3 = zis.getNextEntry()
    //     Assertions.assertThat(entry1.getSize()).isNotZero()
    //     Assertions.assertThat(entry2.getSize()).isNotZero()
    //     Assertions.assertThat(entry3).isNull()

    // }
}

File: uservice-timeentries/main/host/src/test/java/sinnet/reports/ReportService2Test.java
package sinnet.reports;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Timeout;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.ContextConfiguration;

import sinnet.AppOperations;
import sinnet.db.PostgresDbExtension;
import sinnet.host.HostTestContextConfiguration;
import sinnet.report2.grpc.ActivityDetails;
import sinnet.report2.grpc.ReportRequest;
import sinnet.Profiles;

@SpringBootTest
@ContextConfiguration(classes = { HostTestContextConfiguration.class })
@ActiveProfiles(Profiles.TEST)
@ExtendWith(PostgresDbExtension.class)
@Timeout(value = 3)
class ReportService2Test {

  @Autowired
  AppOperations self;

  @Test
  void produceReportWithMinDataCase1() {
    var activity = ActivityDetails.newBuilder().build();
    var request = ReportRequest.newBuilder().build();
    var service = self.getSelfReport2();
    var res = service.produce(request);
    var data = res.getData().toByteArray();
    Assertions
        .assertThat(data)
        .isNotEmpty();
  }

  // @Test
  // def produceReportWithMinDataCase2(): Unit = {
  // var period = YearMonthDTO.newBuilder().setYear(2001).setMonth(1);
  // val request = ReportRequestDTO.newBuilder()
  // .addDetails(ActivityDetailsDTO
  // .newBuilder()
  // .setPersonName("Ala")
  // .setYearMonth(period)
  // .setHowLongInMins(42)
  // .build())
  // .addDetails(ActivityDetailsDTO
  // .newBuilder()
  // .setPersonName("Ola")
  // .setYearMonth(period)
  // .setHowLongInMins(123)
  // .setHowFarInKms(4)
  // .build())
  // .build()
  // val res = self.produce(request)
  // var data = res.getData().toByteArray()

  // // uncomment block of lines below to produce a local example raport file
  // import java.io.File
  // import java.nio.file.Files
  // import java.nio.file.Paths
  // import java.util.concurrent.TimeUnit;

  // import org.junit.jupiter.api.Timeout;
  // import org.junit.jupiter.api.Timeout.ThreadMode;
  // import org.springframework.beans.factory.annotation.Autowired;

  // import sinnet.AppOperations;
  // import sinnet.report2.grpc.ActivityDetails;
  // import sinnet.report2.grpc.ReportRequest;
  // Files.write(Paths.get("temp_raport2_from_test.pdf"), data)

  // Assertions.assertThat(data).isNotEmpty()
  // }

}
